src/expr/impl/src/scalar/timestamptz.rs

```diff
@@ -46,22 +47,24 @@ pub fn timestamp_at_time_zone(input: Timestamp, time_zone: &str) -> Result<Times
     // https://www.postgresql.org/docs/current/datetime-invalid-input.html
     // Special cases:
     // * invalid time during daylight forward
-    //   * PostgreSQL uses UTC offset before the transition
-    //   * We report an error (FIXME)
+    //   * use UTC offset before the transition
     // * ambiguous time during daylight backward
-    //   * We follow PostgreSQL to use UTC offset after the transition
-    let instant_local = input
-        .0
-        .and_local_timezone(time_zone)
-        .latest()
-        .ok_or_else(|| ExprError::InvalidParam {
-            name: "local timestamp",
-            reason: format!(
-                "fail to interpret local timestamp \"{}\" in time zone \"{}\"",
-                input, time_zone
-            )
-            .into(),
-        })?;
+    //   * use UTC offset after the transition
+    let instant_local = match input.0.and_local_timezone(time_zone) {
+        LocalResult::Single(t) => t,
+        LocalResult::None => (input.0 + chrono::Duration::hours(1))
+            .and_local_timezone(time_zone)
+            .latest()
+            .ok_or_else(|| ExprError::InvalidParam {
```

Comment:
I'm not sure if it is safe to unwrap here. To be conservative, I still make it return an error.